did/plugins/sentry.py

```diff
@@ -71,12 +85,19 @@ def filter_data(self):
         stats = []
         log.debug("Query: Date range {0} - {1}".format(
             str(self.options.since.date), str(self.options.until.date)))
-        for activity in self.sentry.get_data():
-            date = self.get_date(activity)
-            if (date >= str(self.options.since.date) and
-                    date <= str(self.options.until.date) and
-                    activity['type'] != "set_regression"):
-                stats.append(activity)
+        next_page = True
+        while next_page:
+            for activity in self.sentry.get_data():
+                date = self.get_date(activity)
+                if (date > str(self.options.until.date)):
+                    continue
+                if (date < str(self.options.since.date)):
+                    next_page = False
+                    break
+                if (activity['type'] != "set_regression"):
+                    stats.append(activity)
+        # erase cursor, so next stats will be search from the start
+        self.sentry.cursor = ''
         return stats
```

Comment:
Wouldn't it be better to completely handle the pagination inside `SentryAPI.get_data()`. In this way we could also easily prevent fetching data from the server multiple times.

Reply:
The problem is plugin does not know how much data should pull out of server, so in order to avoid situation when actual data are 5 entries and the plugin would download whole history I did it this way. This could be moved to ```SentryAPI.get_data()``` method by calling ```SentryAPI.get_data(since, until)``` inside ```SentryStats.filter_data()``` which I realize now.  
  
About the fetching data multiple times, I fetch data multiple times because I created the plugin following the bugzilla plugin example. So when you only want to list ```--sentry-resolved``` the plugin would be searching only for resolved.

Reply:
Yes, you could limit the search by `since` and `until` and do the download just once. Or, if the API supports more detailed search, then let's use it to prevent fetching of unrelated data. That would be even better than fetching all data for the organization.

For sharing the once-fetched data you can get some inspiration in the `nitrate` plugin:

https://github.com/psss/did/blob/master/did/plugins/nitrate.py#L100

See properties `cases` and `copies` for one way how this could be done. But you could store the data in `SentryAPI` as well. No need to have it under the `StatsGroup` instance.